extmod/network_ninaw10.c

```diff
@@ -99,6 +103,40 @@ STATIC void network_ninaw10_poll_sockets(mp_sched_node_t *node) {
     }
 }
 
+STATIC void network_ninaw10_poll_connect(mp_sched_node_t *node) {
+    nina_obj_t *self = &network_nina_wl_sta;
+
+    int status = nina_connection_status();
+    if (status == NINA_STATUS_CONNECTED) {
+        // Connected to AP, nothing else to do.
+        return;
+    }
+
+    if (status != NINA_STATUS_NO_SSID_AVAIL) {
+        // If not connected, and no connection in progress, the connection attempt has failed.
+        // Read the ESP failure reason, reconnect and reschedule the connection polling code.
+        int reason = nina_connection_reason();
+        if (reason == NINA_ESP_REASON_AUTH_EXPIRE ||
+            reason == NINA_ESP_REASON_ASSOC_EXPIRE ||
+            reason == NINA_ESP_REASON_NOT_AUTHED ||
+            reason == NINA_ESP_REASON_4WAY_HANDSHAKE_TIMEOUT ||
+            reason >= NINA_ESP_REASON_BEACON_TIMEOUT) {
```

Comment:
I can't see these constants defined anywhere, and I don't think this code is built by the CI... might be worth adding `ARDUINO_NANO_RP2040_CONNECT` to the rp2 CI to build this nanw10 driver.

Reply:
Yes you're right, they were in `nina_wifi_drv.h` forgot to add it to the commit. 

> might be worth adding ARDUINO_NANO_RP2040_CONNECT to the rp2 CI to build this nanw10 driver.

Would be nice, and it uses other features not used by other boards so would be a good test.

Reply:
Done, I think.